### PR TITLE
Added Purview Role Assignment & Storage Access Rules

### PR DESCRIPTION
**Pull request by marvinbuss**
**This PR fixes**
* Added storage resource access rules for Synapse workspaces and Purview account
* Added Purview role assignment to subscription as Storage Blob Data Rader and Reader to be able to scan all storage accounts and dedicated SQL Pools
* Moved some docs
* Linting

### CODE REVIEW

**Review by marvinbuss — COMMENTED** [review]
LGTM!

**Comment by marvinbuss** [comment]
[Preview Link](https://portal.azure.com/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fdata-landing-zone%2Fmarvinbuss%2Fpurview_integration%2Finfra%2Fmain.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fdata-landing-zone%2Fmarvinbuss%2Fpurview_integration%2Fdocs%2Freference%2Fportal.dataLandingZone.json)

File: docs/guidance/EnterpriseScaleAnalytics-DatabricksManualSetup.md [hunk 1]
```diff
@@ -8,9 +8,9 @@ Due to the following service limitations, Databricks needs to be setup manually:
 
 ## Manual Databricks configuration
 
-Due to the issue mentioned above, we cannot rely on the application workflow, but only rely on the on-behalf workflow. This means, that instead of using a Service Principle for authentication, we need to rely on a user being authenticated against the workspace. Only then, we can use the Databricks API to create a Key Vault backed secret scopes. If you are ok with Databricks backed secret scopes, then you can already automate the complete setup end-to-end. However, for manageability reasons, we are recommending to use Key Vaults for storing secrets.
+Due to the issue mentioned above, we cannot rely on the application workflow, but only rely on the on-behalf workflow. This means, that instead of using a Service Principle for authentication, we need to rely on a user being authenticated against the workspace. Only then, we can use the Databricks API to create a Key Vault backed secret scopes. If you are OK with Databricks backed secret scopes, then you can already automate the complete setup end-to-end. However, for manageability reasons, we are recommending to use Key Vaults for storing secrets.
```
Comment (amanjeetsingh) [inline docs/guidance/EnterpriseScaleAnalytics-DatabricksManualSetup.md:11]: Typo - change "principle" to "principal"

"we are recommending to use Key Vaults for storing secrets." change to "we recommend using Azure Key Vault for storing secrets."

**Review by amanjeetsingh — COMMENTED** [review]
Minor updates; see comments for more details.

**Review by marvinbuss — COMMENTED** [review]
LGTM!

**Review by marvinbuss — COMMENTED** [review]
LGTM!

**Review by sasever — APPROVED** [review]
Good To Go

**Pull request merged**